xenoglossia/builtins.py

```diff
@@ -167,7 +168,22 @@ def flip(input, *args):
     """
     Returns an upside down version of *input* prepended by our favorite Flip Table dude.
     """
-    flipped_input = "".join(_FLIP_CHAR_DICT.get(c, c) for c in input[::-1])
+    char_list = []
+    pair = None
+    for c in input[::-1]:
+        if ord(c) in range(0xDC00, 0xE000):
```

Comment:
This does a linear search in a materialized list on Python 2.  Suggest using `if 0xDC00 <= ord(c) < 0xE000:` instead.